drivers/audio/codec_shell.c

```diff
@@ -0,0 +1,198 @@
+	uint8_t value;
+};
+
+static const struct args_index args_indx = {
+	.device = 1,
+	.property = 2,
+	.channel = 3,
+	.value = 4,
+};
+
+static int parse_named_int(const char *name, const char *keystack[], size_t count)
+{
+	char *endptr;
+	int i;
+
+	/* Attempt to parse name as a number first */
+	i = strtoul(name, &endptr, 0);
+
+	if (*endptr == '\0') {
+		return i;
+	}
+
+	/* Name is not a number, look it up */
+	for (i = 0; i < count; i++) {
+		if (strcmp(name, keystack[i]) == 0) {
+			return i;
+		}
+	}
+
+	return -ENOTSUP;
+}
+
+static int cmd_start(const struct shell *sh, size_t argc, char *argv[])
+{
+	const struct device *dev;
+
+	dev = device_get_binding(argv[args_indx.device]);
+	if (!dev) {
+		shell_error(sh, "Audio Codec device not found");
+		return -ENODEV;
+	}
+	audio_codec_start_output(dev);
+
+	return 0;
+}
+
+static int cmd_stop(const struct shell *sh, size_t argc, char *argv[])
+{
+	const struct device *dev;
+
+	dev = device_get_binding(argv[args_indx.device]);
+	if (!dev) {
+		shell_error(sh, "Audio Codec device not found");
+		return -ENODEV;
+	}
+	audio_codec_stop_output(dev);
+
+	return 0;
+}
+
+static int cmd_set_prop(const struct shell *sh, size_t argc, char *argv[])
+{
+	const struct device *dev;
+	int property;
+	int channel;
+	long value;
+	char *endptr;
+	audio_property_value_t property_value;
+
+	dev = device_get_binding(argv[args_indx.device]);
+	if (!dev) {
+		shell_error(sh, "Audio Codec device not found");
+		return -ENODEV;
+	}
+
+	property = parse_named_int(argv[args_indx.property], codec_property_name,
+				   ARRAY_SIZE(codec_property_name));
+	if (property < 0) {
+		shell_error(sh, "Property '%s' unknown", argv[args_indx.property]);
+		return -EINVAL;
+	}
+
+	channel = parse_named_int(argv[args_indx.channel], codec_channel_name,
+				  ARRAY_SIZE(codec_channel_name));
+	if (channel < 0) {
+		shell_error(sh, "Channel '%s' unknown", argv[args_indx.channel]);
+		return -EINVAL;
+	}
+
+	value = strtol(argv[args_indx.value], &endptr, 0);
+	if (*endptr != '\0') {
+		return -EINVAL;
+	}
+	if (value > INT32_MAX || value < INT32_MIN) {
+		return -EINVAL;
+	}
+	switch (property) {
+	case AUDIO_PROPERTY_OUTPUT_VOLUME:
+		property_value.vol = value;
+		break;
+	case AUDIO_PROPERTY_OUTPUT_MUTE:
+		property_value.mute = value;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return audio_codec_set_property(dev, property, channel, property_value);
+}
+
+static int cmd_apply_prop(const struct shell *sh, size_t argc, char *argv[])
+{
+	const struct device *dev;
+
+	dev = device_get_binding(argv[args_indx.device]);
+	if (!dev) {
+		shell_error(sh, "Audio Codec device not found");
+		return -ENODEV;
+	}
+
+	return audio_codec_apply_properties(dev);
+}
+
+/* Device name autocompletion support */
+static void device_name_get(size_t idx, struct shell_static_entry *entry)
+{
+	const struct device *dev = shell_device_lookup(idx, NULL);
+
+	entry->syntax = (dev != NULL) ? dev->name : NULL;
+	entry->handler = NULL;
```

Comment:
you can go with `entry->syntax = (dev) ? : dev->name : NULL;

Reply:
The same code is used in several (10) others shell files:

https://github.com/zephyrproject-rtos/zephyr/blob/6d67a56d5b5ddb3f744eb844e6b4f7d033be7a94/drivers/sensor/sensor_shell.c#L543

https://github.com/zephyrproject-rtos/zephyr/blob/6d67a56d5b5ddb3f744eb844e6b4f7d033be7a94/drivers/i2c/i2c_shell.c#L314

...

I propose you to leave it as it